demos/coreMQTT/mqtt_demo_mutual_auth.c

```diff
@@ -0,0 +1,996 @@
+ * network connection.
+ *
+ * The Example shown below uses MQTT APIs to create MQTT messages and send them
+ * over the mutually authenticated network connection established with the
+ * MQTT broker. This example is single threaded and uses statically allocated
+ * memory. It uses QoS1 for sending to and receiving messages from the broker.
+ *
+ * A mutually authenticated TLS connection is used to connect to the
+ * MQTT message broker in this example. Define democonfigMQTT_BROKER_ENDPOINT,
+ * democonfigROOT_CA_PEM, democonfigCLIENT_CERTIFICATE_PEM,
+ * and democonfigCLIENT_PRIVATE_KEY_PEM in mqtt_demo_mutual_auth_config.h to establish a
+ * mutually authenticated connection.
+ */
+
+/**
+ * @file mqtt_demo_mutual_auth.c
+ * @brief Demonstrates usage of the MQTT library.
+ */
+
+/* Standard includes. */
+#include <string.h>
+#include <stdio.h>
+
+/* Demo Specific configs. */
+#include "mqtt_demo_mutual_auth_config.h"
+
+#include "aws_demo.h"
+
+/* Kernel includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+
+/* MQTT library includes. */
+#include "core_mqtt.h"
+
+/* Retry utilities include. */
+#include "retry_utils.h"
+
+/* Transport interface implementation include header for TLS. */
+#include "transport_secure_sockets.h"
+
+/* Include header for connection configurations. */
+#include "aws_clientcredential.h"
+
+/* Include header for client credentials. */
+#include "aws_clientcredential_keys.h"
+
+/* Include header for root CA certificates. */
+#include "iot_default_root_certificates.h"
+
+/*------------- Demo configurations -------------------------*/
+
+/** Note: The device client certificate and private key credentials are
+ * obtained by the transport interface implementation (with Secure Sockets)
+ * from the demos/include/aws_clientcredential_keys.h file.
+ *
+ * The following macros SHOULD be defined for this demo which uses both server
+ * and client authentications for TLS session:
+ *   - keyCLIENT_CERTIFICATE_PEM for client certificate.
+ *   - keyCLIENT_PRIVATE_KEY_PEM for client private key.
+ */
+
+#ifndef democonfigMQTT_BROKER_ENDPOINT
+    #define democonfigMQTT_BROKER_ENDPOINT    clientcredentialMQTT_BROKER_ENDPOINT
+#endif
+
+#ifndef democonfigROOT_CA_PEM
+    #define democonfigROOT_CA_PEM    tlsATS1_ROOT_CERTIFICATE_PEM
+#endif
+
+#ifndef democonfigCLIENT_IDENTIFIER
+
+/**
+ * @brief The MQTT client identifier used in this example.  Each client identifier
+ * must be unique so edit as required to ensure no two clients connecting to the
+ * same broker use the same client identifier.
+ */
+    #define democonfigCLIENT_IDENTIFIER    clientcredentialIOT_THING_NAME
+#endif
+
+#ifndef democonfigMQTT_BROKER_PORT
+
+/**
+ * @brief The port to use for the demo.
+ */
+    #define democonfigMQTT_BROKER_PORT    clientcredentialMQTT_BROKER_PORT
+#endif
+
+/**
+ * @brief The maximum number of times to run the subscribe publish loop in this
+ * demo.
+ */
+#ifndef democonfigMQTT_MAX_DEMO_COUNT
```

Comment:
I add this to all the plaintext demo configs as you had suggested

Reply:
I have retained this here as well as added commented section for the config in the `mqtt_demo_mutual_auth_config.h` files (ommit: a85e6555c)